tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,353 @@
+    'verify_target_repo_on_satellite',
+    'precondition_check_upgrade_and_install_leapp_tool',
+    'fix_inhibitors',
+)
+def test_leapp_upgrade_rhel(
+    module_target_sat,
+    custom_leapp_host,
+    upgrade_path,
+):
+    """Test to upgrade RHEL host to next major RHEL Realse with Leapp Preupgrade and Leapp Upgrade
+    Job templates
+
+    :id: 8eccc689-3bea-4182-84f3-c121e95d54c3
+
+    :Steps:
+        1. Import a subscription manifest and enable, sync source & target repositories
+        2. Create LCE, Create CV, add repositories to it, publish and promote CV, Create AK, etc.
+        3. Register content host with AK
+        4. Varify target rhel repositories are enable on Satellite
+        5. Update all packages, install leapp tool and fix inhibitors
+        6. Run Leapp Preupgrade and Leapp Upgrade job template
+
+    :expectedresults:
+        1. Update RHEL OS major version to another major version
+
+    """
+    logger.info('Running test for upgrade path %s', upgrade_path)
+    old_ver = custom_leapp_host.os_version.major
+    # 6. Run LEAPP-PREUPGRADE Job Template-
+    template_id = (
+        module_target_sat.api.JobTemplate()
+        .search(query={'search': 'name="Run preupgrade via Leapp"'})[0]
+        .id
+    )
+    job = module_target_sat.api.JobInvocation().run(
+        synchronous=False,
+        data={
+            'job_template_id': template_id,
+            'targeting_type': 'static_query',
+            'search_query': f'name = {custom_leapp_host.hostname}',
+        },
+    )
+    module_target_sat.wait_for_tasks(
+        f'resource_type = JobInvocation and resource_id = {job["id"]}', poll_timeout=1800
+    )
+    result = module_target_sat.api.JobInvocation(id=job['id']).read()
+    assert result.succeeded == 1
+    # Run LEAPP-UPGRADE Job Template-
+    template_id = (
+        module_target_sat.api.JobTemplate()
+        .search(query={'search': 'name="Run upgrade via Leapp"'})[0]
+        .id
+    )
+    job = module_target_sat.api.JobInvocation().run(
+        synchronous=False,
+        data={
+            'job_template_id': template_id,
+            'targeting_type': 'static_query',
+            'search_query': f'name = {custom_leapp_host.hostname}',
+            'inputs': {'Reboot': 'true'},
+        },
+    )
+    module_target_sat.wait_for_tasks(
+        f'resource_type = JobInvocation and resource_id = {job["id"]}', poll_timeout=1800
+    )
+    result = module_target_sat.api.JobInvocation(id=job['id']).read()
+    assert result.succeeded == 1
+    # Wait for the host to be rebooted and SSH daemon to be started.
+    try:
+        wait_for(
+            custom_leapp_host.connect,
+            fail_condition=lambda res: res is not None,
+            handle_exception=True,
+            raise_original=True,
+            timeout=180,
+            delay=1,
+        )
+    except ConnectionRefusedError:
+        raise ConnectionRefusedError('Timed out waiting for SSH daemon to start on the host')
```

Comment:
Please replace it with 
```suggestion
    custom_leapp_host.wait_for_connection()
```